webhook_server/libs/github_api.py

```
@@ -733,6 +735,30 @@ def _repo_data_from_config(self, repository_config: dict[str, Any]) -> None:
+        _repo_bp = _repo_bp if isinstance(_repo_bp, dict) else {}
+        # Repository-level overrides global; default from DEFAULT_BRANCH_PROTECTION
+        self.required_conversation_resolution: bool = _bp_default
+        for _bp_scope, _bp_dict in [("global", _global_bp), ("repository", _repo_bp)]:
+            if _bp_key in _bp_dict:
+                _bp_val = _bp_dict[_bp_key]
+                if isinstance(_bp_val, bool):
+                    self.required_conversation_resolution = _bp_val
+                else:
+                    self.logger.warning(
+                        f"{self.log_prefix} Invalid branch-protection.{_bp_key} value in {_bp_scope} config: "
+                        f"{_bp_val!r} (expected bool), using default: {_bp_default}"
```

Ruff (0.15.2)

[warning] Logging statement uses f-string

(G004)